libsplinter/src/rest_api/mod.rs

```diff
@@ -933,6 +933,28 @@ impl RestApiBuilder {
                         }
 
                         let oauth_client = match oauth_config {
+                            #[cfg(feature = "oauth-openid")]
+                            OAuthConfig::Azure {
+                                client_id,
+                                client_secret,
+                                redirect_url,
+                                oauth_openid_url,
+                                inflight_request_store,
+                            } => OpenIdOAuthClientBuilder::new()
+                                .with_discovery_url(oauth_openid_url)
+                                .with_client_id(client_id)
+                                .with_client_secret(client_secret)
+                                .with_redirect_url(redirect_url)
+                                .with_inflight_request_store(inflight_request_store)
+                                .build()
+                                .map_err(|err| {
```

Comment:
This is incorrect, as OAuthClientBuildError has multiple variants, not all of which should be mapped to InvalidStateError. (This might suggests OpenIdOAuthClientBuilder.build()'s signature is problematic, though RestApiServerError does have an InternalError variant, so the non-InvalidStateError cases could be mapped to that.) If OAuthClientBuildError::InternalStateError is the only one handled here, it may be better to extract the InternalStateError and insert it into the RestApiServerError without wrapping it, since the additional level of wrapping is probably not useful.

Reply:
Added a new commit to replace the `map_err`s from each branch with a `From<OAuthClientBuildError>` implementation for `RestApiServerError`.